DigitalLearningSolutions.Data/DataServices/ProgressDataService.cs

```diff
@@ -227,5 +235,98 @@ public void UnlockProgress(int progressId)
                 new { progressId }
             );
         }
+
+        public Progress? GetProgressByProgressId(int progressId)
+        {
+            return connection.Query<Progress>(
+                @"SELECT ProgressId,
+                        CandidateID,
+                        CustomisationID,
+                        Completed,
+                        RemovedDate,
+                        SystemRefreshed,
+                        SupervisorAdminID,
+                        CompleteByDate,
+                        EnrollmentMethodID,
+                        EnrolledByAdminID,
+                        SubmittedTime,
+                        CustomisationVersion,
+                        DiagnosticScore
+                    FROM Progress
+                    WHERE ProgressID = @progressId",
+                new { progressId }
+            ).SingleOrDefault();
+        }
+
+        public IEnumerable<DetailedSectionProgress> GetSectionProgressDataForProgressEntry(int progressId)
+        {
+            return connection.Query<DetailedSectionProgress>(
+                @"SELECT
+                        s.SectionID,
+                        s.SectionName,
+                        (SUM(asp1.TutStat) * 100) / (COUNT(t.TutorialID) * 2) AS Completion,
+                        SUM(asp1.TutTime) AS TotalTime,
+                        s.AverageSectionMins AS AverageTime,
+                        CASE WHEN (s.PLAssessPath IS NOT NULL AND cu.IsAssessed = 1) THEN 1 ELSE 0 END AS PostLearningAssessment,
+                        COALESCE (MAX(ISNULL(aa.Score, 0)), 0) AS Outcome,

```

Comment:
I don't think so on examination I don't see how a null value escapes the inner ISNULL. It's inherited from the original SQL queries so I'm loath to remove it, but if we're confident it doesn't do anything I'll cut it.

Reply:
My feelings are that it's fine to copy across the old SQL, rename aliases and remove the columns being returned that we don't want.
However, if we're making bigger changes than that we should pull that logic into C# code so we can label it with a useful method name, reuse it and test it separately. Unless it's going to require us to add a whole load of SQL to pull the ingredient data in from the db instead of calculating in the db.